File: app/src/main/java/com/example/admin/mybledemo/DeviceAdapter.java
package com.example.admin.mybledemo;

import android.content.Context;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.Button;
import android.widget.TextView;
import java.util.List;

import cn.com.heaton.blelibrary.ble.Ble;
import cn.com.heaton.blelibrary.ble.model.BleDevice;

/**
 *
 * Created by LiuLei on 2016/11/26.
 */


public class DeviceAdapter extends BaseAdapter {
    private List<BleDevice> mLeDevices;
    private Context mContext;

    public DeviceAdapter(Context context, List<BleDevice> devices) {
        mLeDevices = devices;
        mContext = context;
    }

    @Override
    public int getCount() {
        return mLeDevices.size();
    }

    @Override
    public BleDevice getItem(int i) {
        return mLeDevices.get(i);
    }

    @Override
    public long getItemId(int i) {
        return i;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        final ViewHolder viewHolder;
        if (view == null) {
            view = LayoutInflater.from(mContext).inflate(R.layout.listitem_device, null);
            viewHolder = new ViewHolder();
            viewHolder.deviceAddress = (TextView) view.findViewById(R.id.device_address);
            viewHolder.deviceName = (TextView) view.findViewById(R.id.device_name);
            viewHolder.deviceState = (TextView) view.findViewById(R.id.state);
            viewHolder.cancelReConnect = (Button) view.findViewById(R.id.cancelReConnect);
            view.setTag(viewHolder);
        } else {
            viewHolder = (ViewHolder) view.getTag();
        }

        final BleDevice device = mLeDevices.get(i);
        final String deviceName = device.getBleName();
        if(device.isConnectting()){
            viewHolder.deviceState.setText("正在连接中...");
        }
        else if(device.isConnected()){
            viewHolder.deviceState.setText("已连接");
        }else {
            viewHolder.deviceState.setText("未连接");
        }
        if (TextUtils.isEmpty(deviceName)){
            viewHolder.deviceName.setText("未知设备");
        }else {
            viewHolder.deviceName.setText(deviceName);
        }

        if (device.isAutoConnect()){
            viewHolder.cancelReConnect.setText("取消重连");
        }else {
            viewHolder.cancelReConnect.setText("重连");
        }
        viewHolder.deviceAddress.setText(device.getBleAddress());
        viewHolder.cancelReConnect.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (device.isAutoConnect()){
                    Ble.getInstance().resetReConnect(device, false);
                    Utils.showToast("已取消重连");
                    viewHolder.cancelReConnect.setText("重连");
                    if (!device.isConnected()){
                        viewHolder.deviceState.setText("未连接");
                    }
                }else {
                    Ble.getInstance().resetReConnect(device, true);
                    Utils.showToast("开启重连");
                    viewHolder.cancelReConnect.setText("取消重连");
                }
            }
        });

        return view;
    }

    class ViewHolder {
        TextView deviceName;
        TextView deviceAddress;
        TextView deviceRSSI;
        TextView deviceState;
        Button cancelReConnect;
    }

}

File: appkotlin/src/main/java/com/i502tech/appkotlin/DeviceAdapter.kt
package com.i502tech.appkotlin

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import cn.com.heaton.blelibrary.ble.model.BleDevice
import kotlinx.android.synthetic.main.item_device.view.*

/**
 * description $desc$
 * created by jerry on 2019/5/4.
 */
class DeviceAdapter(var items: List<BleDevice>) : RecyclerView.Adapter<DeviceAdapter.ViewHolder>() {

    private var mListener: ((Int) -> Unit?)? = null

    fun setOnItemClickListener(mListener: (Int) -> Unit) {
        this.mListener = mListener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_device, parent, false)
        return ViewHolder(v)
    }

    override fun getItemCount(): Int = items.size

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val device = items[position]
        device.run {
            holder.itemView.device_name.text = bleName ?: "未知设备"
            holder.itemView.device_address.text = bleAddress
            when {
                device.isConnected -> holder.itemView.state.text = "已连接"
                device.isConnectting -> holder.itemView.state.text = "正在连接中..."
                else -> holder.itemView.state.text = "未连接"
            }
        }
        holder.itemView.setOnClickListener {
            mListener?.invoke(position)
        }
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}